moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp

```diff
@@ -129,10 +131,9 @@ void plan_execution::PlanWithSensing::displayCostSources(bool flag)
 {
   if (flag && !display_cost_sources_)
     // publisher for cost sources
-    cost_sources_publisher_ =
-        node_handle_.advertise<visualization_msgs::MarkerArray>("display_cost_sources", 100, true);
+    cost_sources_publisher_ = node_->create_publisher<visualization_msgs::msg::MarkerArray>("display_cost_sources",rmw_qos_profile_default);
   else if (!flag && display_cost_sources_)
-    cost_sources_publisher_.shutdown();
+    cost_sources_publisher_.reset();
```

Comment:
Does this really work? I could be wrong but I thought the publishers are maintained by the node and stay alive as long as the node does.

Reply:
I'm not sure, does this help .? [link](https://answers.ros.org/question/280108/ros2rclcpp-threadsafety-of-nodecreate_publisher/)